CONTRIBUTING.md

```diff
@@ -9,24 +9,50 @@ Follow these instructions for contributing new recipes. The Goal headers indicat
+1. (optional) If your recipe idea is unique, open a new ["Recipe Request" issue](https://github.com/JordanMartinez/purescript-cookbook/issues/new?assignees=&labels=recipe-request&template=recipe-request.md&title=). This step aims to prevents two people from working on the same recipe.
 1. Proceed to the next step if you'd like to implement this recipe yourself.
 
 #### Goal 2: Setup a New Recipe's Boilerplate by Copying a Current Similar One
 
 1. Pick an existing recipe to duplicate as a starting point. Here are some suggestions:
-    - `HelloLog` - If your recipe is compatible with both Node.js and web browser backends. _Note:_ Logging to the console **is** supported by **both** backends.
-    - `ReadPrintFileContentsNode` - If your recipe is **only** compatible with the `Node.js` backend.
-    - `DiceCLI` - If your recipe is **only** compatible with the `Node.js` backend **and** requires user interact for testing (e.g. a CLI app). This is to prevent our CI process from blocking while waiting for input during testing.
-    - `WindowPropertiesJs` - If your recipe is **only** compatible with the web browser backend.
-    - If your recipe targets a particular web framework:
-        - `HelloConcur`
-        - `HelloHalogenHooks`
-        - `HelloReactHooks`
+
+    | Web | Node | CLI | Framework | Template |
+    | :-: | :-: | :-: | :-: | --- |
+    | :heavy_check_mark: | :heavy_check_mark: | | | `HelloLog` |
+    | :heavy_check_mark: | :x: | | :x: | `GroceriesJs` |
+    | :x: | :heavy_check_mark: | :x: | | `ReadPrintFileContentsNode` |
+    | :x: | :heavy_check_mark: | :heavy_check_mark: | | `DiceCLI` |
+    | :heavy_check_mark: | | | Concur | `HelloConcur` |
+    | :heavy_check_mark: | | | Halogen | `HelloHalogenHooks` |
+    | :heavy_check_mark: | | | React | `HelloReactHooks` |
+
+    Notes on interpreting the table:
+    - An empty cell means "not possible".
+    - **Web** - Recipe is compatible with the web browser backend.
+    - **Node** - Recipe is compatible with the Node.js backend.
+    - **CLI** - Recipe does not terminate, and/or requires user interaction (e.g. a CLI app) when run in the Node.js environment. These recipes contain a special configuration flag to prevent our CI process from stalling on these recipes during testing.
+    - **Framework** - Recipe uses a particular web framework.
+    - **Template** - A recommended recipe to copy.
+
 1. Run the recipe creation script. Substitute `MyNewRecipe` with your recipe name:
     ```
     ./scripts/newRecipe.sh MyNewRecipe HelloLog
     ```
+    - **Tip:** You may point to the recipe path to take advantage of shell autocompletion
+      ```
+      ./scripts/newRecipe.sh MyNewRecipe recipes/HelloLog
+      ```
+    - **Tip:** The `make` commands printed after running this script will be helpful when developing your recipe. Make a copy of these for reference.
```

Comment:
Perhaps we should add a command like `make help` that prints out these commands? If a person forgets to make note of them, they can just see it again by running that. The output would replace all instances of a recipe name with `<recipe-name>`

Reply:
Created #231 to track